compiler/ccgcalls.nim

```diff
@@ -787,7 +792,7 @@ proc genNamedParamCall(p: BProc, ri: PNode, d: var TLoc) =
       pl.add("]")
       if d.k == locNone: getTemp(p, typ[0], d)
       assert(d.t != nil)        # generate an assignment to d:
-      var list: TLoc
+      var list: TLoc = default(TLoc)
```

Comment:
Instead `initLoc` should use an `out` parameter.

Reply:
![image](https://github.com/nim-lang/Nim/assets/43030857/7b03eaf2-fcfa-444b-b168-e542e8cba742)

```nim
proc initLoc(result: out TLoc, k: TLocKind, lode: PNode, s: TStorageLoc, flags: TLocFlags = {}) =
  result = TLoc(k:k, storage: s, lode: lode, r: "", flags: flags)
```

Weird, I make `initLoc` and `initLocExpr` use `out` parameters, but it gives a warning in `initLocExpr`.

> Error: Cannot prove that 'result' is initialized. This will become a compile time error in the future. [ProveInit]